File: decl-ts/tests/validation_test.ts
// The fixture corpus (tests/validation, tests/validation/README.md) judged
// by `decl validate <dir>` (src/conformance.ts): every fixture parses,
// checks, and evaluates as its header says.
import { spawnSync } from 'node:child_process';
import { check, total, root, cli } from './common/check.ts';

console.log('== validation: the fixture corpus ==');
const r = spawnSync(process.execPath, [cli, 'validate', 'tests/validation'], {
  encoding: 'utf8',
  cwd: root,
});
check('every fixture judged as its header says', r.status === 0, r.stderr.slice(-400));
check('the summary counts no failure', /^\d+ ok, 0 failed$/m.test(r.stderr), r.stderr.slice(-200));
total();

File: decl-rs/tests/validation_test.rs
//! The fixture corpus (tests/validation, tests/validation/README.md) judged
//! by `decl validate <dir>` (src/conformance.rs): every fixture parses,
//! checks, and evaluates as its header says.
mod common;
use common::*;
use std::process::Command;

#[test]
fn validation_corpus() {
    let out = Command::new(env!("CARGO_BIN_EXE_decl"))
        .args(["validate", "tests/validation"])
        .current_dir(root())
        .output()
        .unwrap();
    let err = String::from_utf8_lossy(&out.stderr);
    assert!(
        out.status.success(),
        "every fixture judged as its header says:\n{err}"
    );
    assert!(
        err.lines().any(|l| l.ends_with(" ok, 0 failed")),
        "the summary counts no failure: {err}"
    );
}
